tests/asgi/test_sse.py

```diff
@@ -182,6 +183,49 @@ async def _test():
     falcon.invoke_coroutine_sync(_test)
 
 
+class TestSerializeJson:
+    @pytest.fixture
+    def client(self):
+        class SomeResource:
+            async def on_get(self, req, resp):
+                async def emitter():
+                    yield SSEvent(json={'foo': 'bar'})
+                    yield SSEvent(json={'bar': 'baz'})
+
+                resp.sse = emitter()
+
+        resource = SomeResource()
+
+        app = App()
+        app.add_route('/', resource)
+
+        client = testing.TestClient(app)
+        return client
+
+    def test_use_media_handler_dumps(self, client):
+        h = client.app.resp_options.media_handlers[falcon.MEDIA_JSON]
+        h.dumps = lambda x: json.dumps(x).upper()
+
+        result = client.simulate_get()
+        assert result.text == (
+            'data: {"FOO": "BAR"}\n'
+            '\n'
+            'data: {"BAR": "BAZ"}\n'
+            '\n'
+        )
+
+    def test_no_json_media_handler(self, client):
+        client.app.resp_options.media_handlers.pop(falcon.MEDIA_JSON)
```

Comment:
I find it hard to reason about the effects of this `pop` before https://github.com/falconry/falcon/pull/1785 is merged.
But given the test passes, we can probably leave it as-is.